source/Halibut.Tests/TraceLogFileLogger.cs

```diff
@@ -1,25 +1,82 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
 
 namespace Halibut.Tests
 {
     public class TraceLogFileLogger : IDisposable
     {
-        string tempFilePath = Path.GetTempFileName();
+        readonly AsyncQueue<string> queue = new AsyncQueue<string>();
+        readonly string tempFilePath = Path.GetTempFileName();
         string testHash;
 
+        readonly TaskCompletionSource<bool> noMoreDelaysToWritingLogs;
+        readonly Task noMoreDelaysTask;
+        readonly Task writeDataToDiskTask;
+
+        public TraceLogFileLogger()
+        {
+            noMoreDelaysToWritingLogs = new TaskCompletionSource<bool>();
+            noMoreDelaysTask = noMoreDelaysToWritingLogs.Task;
+            writeDataToDiskTask = WriteDataToFile();
+        }
+
         public void SetTestHash(string testHash)
         {
             this.testHash = testHash;
         }
         
         public void WriteLine(string logMessage)
         {
-            File.AppendAllLines(tempFilePath, new[] { logMessage });
+            queue.Enqueue(logMessage);
+        } 
+        
+        async Task WriteDataToFile()
+        {
+            while (true)
+            {
+                // Don't hammer the disk, let some log message queue up before writing them.
+                if (!noMoreDelaysTask.IsCompleted)
+                {
+                    await Task.WhenAny(Task.Delay(5), noMoreDelaysTask);
+                }
+
+                var list = new List<string>();
+                list.Add(await queue.DequeueAsync());
+                while (queue.TryDequeue(out var log))
+                {
+                    list.Add(log);
+                }
+
+                using (var fileAppender = new StreamWriter(tempFilePath, true, Encoding.UTF8, 8192))
```

Comment:
default buffer size is 1KB, i chose 8KB to try and minimise disk writes which tents to be very slow on windows VMs.

Reply:
Is the idea here that there are multiple threads adding log entries to the queue but only one thread dequeueing and writing?